### PR TITLE
LightGraphs -> Graphs, conform with Base interface

### PR DESCRIPTION
**Pull request by April-Hannah-Lena**
This is just a dependency change from LightGraphs.jl to Graphs.jl, and some changes to conform with the functions in Julia Base. For example: `keytype(::BoxPartition)` is changed to `Base.keytype(::BoxPartition)`, `keys_all` changed to `Base.keys`, etc.

There is now also a boundscheck performed when using `union(b::BoxSet...)` which tests if the partitions from the BoxSets are the same. As with Base Julia boundschecks, you can disable it by using `@inbounds union(b::BoxSet...)`.

### CODE REVIEW

File: examples/recurrent_set.jl [hunk 1]
```diff
@@ -80,7 +80,7 @@ function interpolate_v(v, a, b, n, tol)
 end
 
 vf = interpolate_v(knotted_v, -3.5, 3.5, 30, 1e-12)
-f(x) = rk4_flow_map(vf, x, step_size = 0.075)
+f(x) = rk4_flow_map(vf, x, 0.075)
```
Comment (gaioguy) [inline examples/recurrent_set.jl:83]: I would actually rather keep the named parameter here.

File: src/algorithms.jl [hunk 2]
```diff
@@ -113,7 +113,7 @@ const half, sixth, third = 1/2, 1/6, 1/3
     return @. x + τ * dx
 end
 
-@propagate_inbounds function rk4_flow_map(v, x; step_size=0.01, steps=20)
+@inline function rk4_flow_map(v, x, step_size=0.01, steps=20)
```
Comment (gaioguy) [inline src/algorithms.jl:116]: ... see above.

**Review by gaioguy — APPROVED** [review]

File: examples/recurrent_set.jl (hunk 1 above)
Comment (April-Hannah-Lena) [reply examples/recurrent_set.jl:83]: The remov of keyword args was a performance decision, I think we only briefly talked about it before. For some reason, defining the function with keywords increases allocations dramatically. For example in `unstable_set.jl` one van see the allocations jump from O(1) to O(n), where n is the number of test points.

**Comment by April-Hannah-Lena** [comment]
I should have said, this PR was meant to be merged before the CUDA PR, since these changes were already in use in the Cuda version (except for the transition to Graphs.jl). I'll close this PR and reopen with the updated master branch

File: examples/recurrent_set.jl (hunk 1 above)
Comment (gaioguy) [reply examples/recurrent_set.jl:83]: Ok, wow, then we go without keywords, of course ...

**Pull request closed without merging**